piptools/resolver.py

```diff
@@ -185,11 +189,7 @@ def get_best_match(self, ireq):
             Flask==0.10.1 => Flask==0.10.1
 
         """
-        if ireq.editable:
-            # NOTE: it's much quicker to immediately return instead of
-            # hitting the index server
-            best_match = ireq
-        elif is_pinned_requirement(ireq):
+        if ireq.editable or ireq.link or is_pinned_requirement(ireq):
```

Comment:
It used `best_match = ireq` for both `ireq.editable` and `is_pinned_requirement(ireq)`.  This hunk adds `ireq.link` to it, which includes everything to be downloaded.
I cannot remember if that is correct/required really from the top of my head now, sorry.
Maybe it should also use `Link.is_artifact`?!